LibreNMS/Alert/Transport/Gotify.php

```diff
@@ -0,0 +1,103 @@
+<?php
+/* Copyright (C) 2014 Daniel Preussker <f0o@devilcode.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>. */
+
+/**
+ * Gotify API Transport
+ * @author Viktoria Rei Bauer <vbauer@stargazer.at>
+ * @copyright 2019 ToeiRei, f0o, LibreNMS
+ * @license GPL
+ * @package LibreNMS
+ * @subpackage Alerts
+ */
+namespace LibreNMS\Alert\Transport;
+
+use LibreNMS\Alert\Transport;
+
+class Gotify extends Transport
+{
+    public function deliverAlert($obj, $opts)
+    {
+        if (empty($this->config)) {
+            return $this->deliverAlertOld($obj, $opts);
+        }
+        $gotify_server = $this->config['gotify_server'];
+        $gotify_token  = $this->config['gotify_token'];
+        return $this->contactGotify($obj, $gotify_server, $gotify_token);
+    }
+
+    private function deliverAlertOld($obj, $opts)
+    {
+
+        $this->contactGotify($obj, $gotify_server, $gotify_token);
```

Comment:
This function uses $gotify_server and token without defining them. 
But as this is the "legacy" way of sending alerts (without the transport structure) I would say we could just avoid this function and just return false (should not be used anymore). 
@murrant : What do you think ?

Reply:
Yeah, a lot of older code was structured that way because it was modified. (I started a PR to change all that, but never sent it).  You can delete deliverAlertOld and any use of it.

Reply:
I used some existing transport and it had it. I'm completely open to corrections here